tests/components/plex/test_init.py

```diff
@@ -87,91 +60,38 @@ async def test_setup_config_entry_with_error(hass):
     assert entry.state == ENTRY_STATE_SETUP_ERROR
 
 
-async def test_setup_with_insecure_config_entry(hass):
+async def test_setup_with_insecure_config_entry(hass, entry, setup_plex_server):
     """Test setup component with config."""
-
-    mock_plex_server = MockPlexServer()
-
     INSECURE_DATA = copy.deepcopy(DEFAULT_DATA)
     INSECURE_DATA[const.PLEX_SERVER_CONFIG][CONF_VERIFY_SSL] = False
+    entry.data = INSECURE_DATA
 
-    entry = MockConfigEntry(
-        domain=const.DOMAIN,
-        data=INSECURE_DATA,
-        options=DEFAULT_OPTIONS,
-        unique_id=DEFAULT_DATA["server_id"],
-    )
-
-    with patch("plexapi.server.PlexServer", return_value=mock_plex_server), patch(
-        "plexapi.myplex.MyPlexAccount", return_value=MockPlexAccount()
-    ), patch("homeassistant.components.plex.PlexWebsocket.listen") as mock_listen:
-        entry.add_to_hass(hass)
-        assert await hass.config_entries.async_setup(entry.entry_id)
-        await hass.async_block_till_done()
-
-    assert mock_listen.called
+    await setup_plex_server(config_entry=entry)
 
     assert len(hass.config_entries.async_entries(const.DOMAIN)) == 1
     assert entry.state == ENTRY_STATE_LOADED
 
 
-async def test_unload_config_entry(hass):
+async def test_unload_config_entry(hass, entry, mock_plex_server):
     """Test unloading a config entry."""
-    mock_plex_server = MockPlexServer()
-
-    entry = MockConfigEntry(
-        domain=const.DOMAIN,
-        data=DEFAULT_DATA,
-        options=DEFAULT_OPTIONS,
-        unique_id=DEFAULT_DATA["server_id"],
-    )
-    entry.add_to_hass(hass)
-
     config_entries = hass.config_entries.async_entries(const.DOMAIN)
     assert len(config_entries) == 1
     assert entry is config_entries[0]
-
-    with patch("plexapi.server.PlexServer", return_value=mock_plex_server), patch(
-        "plexapi.myplex.MyPlexAccount", return_value=MockPlexAccount()
-    ), patch("homeassistant.components.plex.PlexWebsocket.listen") as mock_listen:
-        assert await hass.config_entries.async_setup(entry.entry_id)
-        await hass.async_block_till_done()
-        assert mock_listen.called
-
     assert entry.state == ENTRY_STATE_LOADED
 
     server_id = mock_plex_server.machineIdentifier
     loaded_server = hass.data[const.DOMAIN][const.SERVERS][server_id]
-
     assert loaded_server.plex_server == mock_plex_server
 
-    with patch("homeassistant.components.plex.PlexWebsocket.close") as mock_close:
-        await hass.config_entries.async_unload(entry.entry_id)
-        assert mock_close.called
-
+    websocket = hass.data[const.DOMAIN][const.WEBSOCKETS][server_id]
+    await hass.config_entries.async_unload(entry.entry_id)
+    assert websocket.close.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6212145)